File: 0804-unique-morse-code-words/0804-unique-morse-code-words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        string mCode[26] = {".-","-...","-.-.","-..","."
            ,"..-.","--.","....","..",".---","-.-",".-..","--"
            ,"-.","---",".--.","--.-",".-.","...","-","..-","...-"
            ,".--","-..-","-.--","--.."};
        unordered_set<string> st;
        for(string s : words){
            string tmp = "";
            for(int i = 0; i < s.size(); i++){
                tmp += mCode[s[i] - 'a'];
            }
            st.insert(tmp);
        }
        return st.size();
    }
};

File: 0804-unique-morse-code-words/0804-unique-morse-code-words.cs
public class Solution {
    public int UniqueMorseRepresentations(string[] words) {
        string[] mCode = {".-","-...","-.-.","-..",".","..-.","--."
            ,"....","..",".---","-.-",".-..","--","-.","---",".--."
            ,"--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        SortedSet<string> st = new();
        foreach(string s in words){
            StringBuilder sb = new();
            foreach(char c in s){
                sb.Append(mCode[c - 'a']);
            }
            st.Add(sb.ToString());
        }
        return st.Count;
    }
}